### PR TITLE
Sdk update to use glm 0.6.1

### PR DESCRIPTION
**Pull request by mayureshagashe2105**
1. Update dependency of `glm` package to use [`0.6.1`](https://github.com/googleapis/google-cloud-python/releases/tag/google-ai-generativelanguage-v0.6.1).
2. proto def for `GenerateContentRequest` lists `system_instruction` in singular as the field is not repeated. This is consistent with the latest release(glm==0.6.1)
3. Import `TypedDict` from `typing_extensions`

## Checklist

- [x] I have performed a self-review of my code.
- [x] I have added detailed comments to my code where applicable.
- [x] I have verified that my change does not break existing code.
- [x] My PR is based on the latest changes of the main branch (if unsure, please run `git pull --rebase upstream main`).
- [x] I am familiar with the [Google Style Guide](https://google.github.io/styleguide/) for the language I have coded in.
- [x] I have read through the [Contributing Guide](https://github.com/google/generative-ai-python/blob/main/CONTRIBUTING.md) and signed the [Contributor License Agreement](https://cla.developers.google.com/about).

### CODE REVIEW

**Review by markmcd — APPROVED** [review]
I added tests & fixed the plural naming.

File: google/generativeai/types/content_types.py
[the remark this answers is not in the document]
Comment (mayureshagashe2105) [reply google/generativeai/types/content_types.py:1]: Oops! `function_calling_mode` was supposed to be a follow-on PR, pushed on the same branch by mistake. Should I revert ?

[the remark this answers is not in the document]
Comment (markmcd) [reply google/generativeai/types/content_types.py:1]: Yeah if you can - push it ASAP & we can merge this.

[the remark this answers is not in the document]
Comment (mayureshagashe2105) [reply google/generativeai/types/content_types.py:1]: Done. Should be good to go.

**Review by markmcd — APPROVED** [review]

File: setup.py [hunk 1]
```diff
@@ -42,7 +42,7 @@ def get_version():
     release_status = "Development Status :: 5 - Production/Stable"
 
 dependencies = [
-    "google-ai-generativelanguage@https://storage.googleapis.com/generativeai-downloads/preview/ai-generativelanguage-v1beta-py.tar.gz",
+    "google-ai-generativelanguage==0.6.1",
```
Comment (MarkDaoust) [inline setup.py:45]: Yes. Thanks!

**Pull request merged**